src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
> evanrolfe 12 minutes ago
> but its more the general mentality of thinking that we should test every tiny detail that I don't think we should get in to the habit.

My thinking is more that, if we care enough to test it, we can also do the extra effort of testing it properly. It's not like this would require hours of development^^.

btw. if @Ana06 wouldn't have requested changes, I would have merged this one now.